### DISCUSSION TITLE
How to retrieve a JS variable list from another file?

### DISCUSSION THREAD

**Discussion by ghost** [community/community#21560] [Other Feature Feedback, Questions, & Ideas]
<p>Hello!</p>
<p>My GitHub page has a small segement of code that writes a random quote from a list:</p>
<pre><code>var quotes = [
    "Quote 1",
    "Quote 2"
]
// Quotes was altered from the original for the sake of this example

var qrandom = Math.floor(Math.random() * (quotes.length));

function randomQuote()
{
      document.getElementById('quote').innerHTML = quotes[qrandom];
}
</code></pre>
<p>Now, could it be possible to grab the “var quotes list” from another JavaScript/JSON/etc. file? If so, how would the code exactly look like?</p>
<p>Thanks.</p>

**Comment by Vic-ST** [comment, marked as answer]
<p>If I understood your question correctly then my answer is…</p>
<p>Here are three files test.js, script.js and index.html</p>
<p>test.js</p>
<pre><code>var qrandom = Math.floor(Math.random() * (quotes.length));

function randomQuote()
{
      document.getElementById('quote').innerHTML = quotes[qrandom];
}

randomQuote();
</code></pre>
<p>script.js</p>
<pre><code>var quotes = [
    "Quote 1",
    "Quote 2"
]
</code></pre>
<p>and index.html</p>
<p>[blank file]</p>
<p>What you would do is add this code to the HTML page:</p>
<pre><code>&lt;script src="script.js"&gt;&lt;/script&gt;
&lt;script src="test.js"&gt;&lt;/script&gt;

&lt;div id="quote"&gt;&lt;/div&gt;
</code></pre>
<p>Please tell me if I misunderstood your question.</p>

**Comment by ghost** [comment]
<p>That’ll work. Thank you.</p>
